File: AlgoAgentXAPI/app/schemas/notifications.py
from datetime import datetime
from typing import Any, Dict, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class NotificationBase(BaseModel):
    type: str = Field(..., description="Notification type")
    title: str
    message: str
    severity: str = "info"
    entity_type: Optional[str] = None
    entity_id: Optional[str] = None
    action_url: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class NotificationCreate(NotificationBase):
    pass


class NotificationUpdate(BaseModel):
    is_read: bool


class NotificationResponse(NotificationBase):
    id: UUID
    user_id: str
    is_read: bool
    created_at: datetime
    read_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class UnreadCountResponse(BaseModel):
    unread_count: int


class MarkReadRequest(BaseModel):
    notification_ids: list[UUID]


class MarkAllReadRequest(BaseModel):
    pass

File: AlgoAgentXApp/src/types/notifications.ts
import { UUID } from "crypto";

export interface NotificationBase {
  type: string;
  title: string;
  message: string;
  metadata?: Record<string, any>;
}

export interface NotificationCreate extends NotificationBase {
  // Additional fields for creation if needed
}

export interface NotificationUpdate {
  is_read: boolean;
}

export interface NotificationResponse extends NotificationBase {
  id: UUID;
  user_id: string;
  is_read: boolean;
  created_at: string; // ISO date string
}

export interface MarkReadRequest {
  notification_ids: UUID[];
}

export interface MarkAllReadRequest {
  // Empty interface for mark all read
}

export interface UnreadCountResponse {
  unread_count: number;
}